### PR TITLE
docs: add metrics doc on barrier

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Credit goes to @BugenZhao for most of the content here.
I have just added some tweaks.

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by lmatz — APPROVED** [review]

File: docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+When injecting a new barrier,
+there will usually be streaming messages in the stream graph (unless it's the initial barrier).
+Since we keep total order for streaming messages,
+this means that all streaming messages currently in the stream graph have to be processed
+before the barrier can pass through.
+If barrier latency is high, it could mean a long time is taken to process these streaming messages.
+Concretely, here are some costs of processing streaming messages:
+1. CPU cost of evaluating expressions.
+2. I/O remote exchange between fragments.
+3. hash-join / hash-agg cache-miss (results in extra costs to access state on s3).
```
Comment (BugenZhao) [inline docs/metrics.md:30]: What about making this statement more general? "Executor" should be enough.

File: docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Next, high barrier latency could also be caused by buffers in the graph.
```
Comment (BugenZhao) [inline docs/metrics.md:32]: May give a hint that in this case we can tweak the buffer size through configuration file (developer), and a heuristic algorithm is on the way. 🤤

**Review by BugenZhao — APPROVED** [review]
Thanks for this doc!

File: docs/metrics.md (hunk 2 above)
Comment (kwannoel) [reply docs/metrics.md:32]: > May give a hint that in this case we can tweak the buffer size through configuration file (developer),
Which options can configure this? So far I just see `stream_connector_message_buffer_size`, but that's just for connector source -> source executor.

> and a heuristic algorithm is on the way. 🤤
Any discussions / issues / RFCs I can link to?

Comment (BugenZhao) [reply docs/metrics.md:32]: > Which options can configure this? So far I just see `stream_connector_message_buffer_size`, but that's just for connector source -> source executor.

https://github.com/risingwavelabs/risingwave/blob/a36e01307d60491b91870ac5a37049a378fe986f/src/config/example.toml#L49-L50

> Any discussions / issues / RFCs I can link to?

I guess we can link to #8654.

Comment (kwannoel) [reply docs/metrics.md:36]: Updated, PTAL

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**